### PR TITLE
Replaced bash script in gCNV ScatterIntervals task with updated version of IntervalListTools.

### LINKED ISSUE #5174: Upgrade picard version and gCNV WDLs after scatter-by-interval is added to IntervalListTools.
See https://github.com/broadinstitute/picard/pull/1208.

### PR DESCRIPTION
**Pull request by samuelklee**
Closes #5174.

Some relevant discussion in #5392.

### CODE REVIEW

**Comment by samuelklee** [comment]
@sooheelee take note of this one as well.

File: scripts/cnv_wdl/cnv_common_tasks.wdl [hunk 1]
```diff
@@ -292,16 +294,35 @@ task ScatterIntervals {
...
-
-        grep @ ${interval_list} > header.txt
-        grep -v @ ${interval_list} > all_intervals.txt
-        split -l ${num_intervals_per_scatter} --numeric-suffixes all_intervals.txt ${base_filename}.scattered.
-        for i in ${base_filename}.scattered.*; do cat header.txt $i > $i.interval_list; done
+        mkdir ${output_dir_}
+        export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk4_jar_override}
+        
+        {
+            >&2 echo "Attempting to run IntervalListTools..."
```
Comment (asmirnov239) [inline scripts/cnv_wdl/cnv_common_tasks.wdl]: Any reason you're redirecting this to stderr? Should we do this for other tasks?

File: scripts/cnv_wdl/cnv_common_tasks.wdl [hunk 2]
```diff
@@ -292,16 +294,35 @@ task ScatterIntervals {
...
+                --SUBDIVISION_MODE INTERVAL_COUNT \
+                --SCATTER_CONTENT ${num_intervals_per_scatter} \
+                --OUTPUT ${output_dir_} &&
+            # output files are named output_dir_/temp_0001_of_N/scattered.interval_list, etc. (N = num_intervals_per_scatter);
+            # we rename them as output_dir_/base_filename.scattered.0000.interval_list, etc.
+            ls ${output_dir_}/*/scattered.interval_list | \
+                cat -n | \
+                while read n filename; do mv $filename ${output_dir_}/${base_filename}.scattered.$(printf "%04d" $n).interval_list; done
+        } || {
+            # if only a single shard is required, then we can just rename the original interval list
```
Comment (asmirnov239) [inline scripts/cnv_wdl/cnv_common_tasks.wdl]: Is it possible to handle this without waiting for IntervalListTools to fail?

File: scripts/cnv_wdl/cnv_common_tasks.wdl (hunk 1 above)
Comment (samuelklee) [reply scripts/cnv_wdl/cnv_common_tasks.wdl]: Cromwell directs GATK logs to stderr as well...just wanted to get all the messages in the same place.

File: scripts/cnv_wdl/cnv_common_tasks.wdl (hunk 2 above)
Comment (samuelklee) [reply scripts/cnv_wdl/cnv_common_tasks.wdl]: Also, this obviates the need to duplicate the logic for calculating SCATTER_COUNT in IntervalListTools (specifically, the decision for what to do when the last scatter block is smaller than the block size), which is desired

**Comment by samuelklee** [comment]
Thanks @asmirnov239, let me know if you approve of my responses!

**Comment by samuelklee** [comment]
@asmirnov239 let me know if this can go in.

File: scripts/cnv_wdl/cnv_common_tasks.wdl (hunk 1 above)
Comment (asmirnov239) [reply scripts/cnv_wdl/cnv_common_tasks.wdl]: Hm, my python logs go to the stdout.log file, at least in Firecloud

**Comment by asmirnov239** [comment]
Looks good @slee!

File: scripts/cnv_wdl/cnv_common_tasks.wdl (hunk 1 above)
Comment (samuelklee) [reply scripts/cnv_wdl/cnv_common_tasks.wdl]: Actually, I think I misspoke above---GATK logs go to stderr regardless.  Currently, gCNV python output goes to stdout (see #4629).  However, note that the messages above are just echoed from bash.

**Pull request merged**